taui/src/components/select-account.js

```diff
@@ -66,61 +57,46 @@ export default class SelectAccount extends PureComponent<Props> {
 
     Storage.put(key, JSON.stringify(profile))
       .then(result => {
-        console.log(result)
-        search() // refresh results
+        search() // Refresh results; will find and go to the new profile
+      })
+      .catch(err => {
+        console.error('Failed to post new profile to S3')
+        console.error(err)
+        this.setState({errorMessage: 'Accounts.CreateError'})
       })
-      .catch(err => console.error(err))
   }
 
   search () {
-    this.setState({errorMessage: ''})
-    const searchName = this.state.headOfHousehold.toUpperCase()
-    const searchVoucher = this.state.voucherNumber.toUpperCase()
+    // Capitalize and strip whitespace from voucher numbers to normalize
+    const searchVoucher = this.state.voucherNumber.toUpperCase().replace(/\s+/g, '')
+    if (!searchVoucher) {
+      this.setState({errorMessage: message('Accounts.SearchError')})
+      return
+    } else {
+      this.setState({errorMessage: ''})
+    }
 
-    const accounts = []
+    var found = false
     Storage.list('')
```

Comment:
Not really related to this PR but something that I thought of while reading this: is there a chance that this filtering approach could get bogged down if S3 gets loaded up with several hundred or thousand profiles? If so, would it be feasible to switch to attempting `Storage.get(searchVoucher)` and then branching based on whether that works or returns a not-found?

I don't think this is something we need to address immediately but if it could become an issue then it would be good to have thought through a solution ahead of time.

Reply:
The initial prototype won't see that level of usage, which is why I went with this simple directory listing search, although I originally expected this page to provide inexact matches (actually search). Since we've decided to limit this to exact matches, we could replace this with a simple `get`.

Reply:
Got it, that makes sense, thanks! I don't think we need to address this right now but it's good to know that we can.

Reply:
Created #66. We'll want to distinguish between file not existing and other `get` error states.